mmv1/products/vertexai/Index.yaml

```diff
@@ -131,6 +131,15 @@ properties:
               performed. Exact reordering is a procedure where results returned by an
               approximate search algorithm are reordered via a more expensive distance computation.
               Required if tree-AH algorithm is used.
+          - !ruby/object:Api::Type::String
+            name: 'shardSize'
+            description: |-
+              Index data is split into equal parts to be processed. These are called "shards".
+              The shard size must be specified when creating an index. The value must be one of the followings:
+              * SHARD_SIZE_SMALL: Small (2GB)
+              * SHARD_SIZE_MEDIUM: Medium (20GB)
+              * SHARD_SIZE_LARGE: Large (50GB)
+            default_from_api: true
```

Comment:
we can have both `default_from_api` and `immutable` be true simultaneously. We should do that if the field is immutable

Reply:
Oops, I mistakenly understood the following error message somehow. I'll restore `immutable: true`. Thank you for your quick catch!

```
/Users/shotarok/ghq/github.com/shotarok/magic-modules/mmv1/api/type.rb:251:in `validate': 'default_value' and 'default_from_api' cannot be both set (RuntimeError)
```

Reply:
I updated it in [b36bed6](https://github.com/GoogleCloudPlatform/magic-modules/pull/8293/commits/b36bed69838e3b86eef510e85e10b72085f6ec00).